frontend/web/components/modals/CreateSegment.tsx

```diff
@@ -319,49 +319,51 @@ const CreateSegment: FC<CreateSegmentType> = ({
   }, [operators, rules])
   //Find any non-deleted rules
   const hasNoRules = !rules[0]?.rules?.find((v) => !v.delete)
-
+  const rulesToShow = rules[0].rules.filter((v) => !v.delete)
```

Comment:
Feel free to ignore this if I'm wrong - it just caught my attention that we're guarding against `rules[0]` being undefined or null on the line above, but not this one.